File: 78-subsets/78-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> lst = new ArrayList<>();
        dfs(0, nums, res, lst);
        return res;
    }
    public void dfs(int idx, int[] nums, List<List<Integer>> res, List<Integer> list) {
        res.add(new ArrayList<>(list));
        for(int i = idx; i < nums.length; i++) {
            list.add(nums[i]);
            dfs(i + 1, nums, res, list);
            list.remove(list.size() - 1);
        }
    }
}

File: 78-subsets/78-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res=[]
        def backtrack(i: int, lst: List[int]):
            nonlocal res
            res.append(lst[:])
            for idx in range(i,len(nums)):
                lst.append(nums[idx])
                backtrack(idx + 1, lst)
                lst.pop()
        backtrack(0,[])
        return res
                